hostlabels/syncer.go

```diff
@@ -0,0 +1,85 @@
+package hostlabels
+
+import (
+	"fmt"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/rancher/go-rancher-metadata/metadata"
+	"github.com/rancher/kubernetes-agent/kubernetesclient"
+)
+
+type hostLabelSyncer struct {
+	kClient        *kubernetesclient.Client
+	metadataClient *metadata.Client
+}
+
+func (h *hostLabelSyncer) syncHostLabels(version string) {
+	err := sync(h.kClient, h.metadataClient)
+	if err != nil {
+		log.Errorf("Error syncing host labels: [%v]", err)
+	}
+}
+
+func sync(kClient *kubernetesclient.Client, metadataClient *metadata.Client) error {
+	hosts, err := metadataClient.GetHosts()
+	if err != nil {
+		log.Errorf("Error reading host list from metadata service: [%v], retrying", err)
+		return err
+	}
+	for _, host := range hosts {
+		node, err := kClient.Node.ByName(host.Hostname)
+		if err != nil {
+			log.Errorf("Error getting node: [%s] by name from kubernetes, err: [%v]", host.Hostname, err)
+			// This node might not have been added to kuberentes cluster yet, so skip it
+			continue
+		}
+		if node.Metadata.Annotations == nil {
+			node.Metadata.Annotations = make(map[string]interface{})
+		}
+		rancherLabelsMetadataStore := node.Metadata.Annotations
+		changed := false
+		//check for new/updated labels
+		for k, v1 := range host.Labels {
+			v2, ok := node.Metadata.Labels[k]
```

Comment:
I'm concerned that this will be too abusive to k8s.  Metadata changes a lot so imagine I have 100 hosts.  We will be making like 100 calls a second to k8s as metadata changes.  It seems like once we sync up with k8s we should keep a list of host->label in memory and only if metadata actually changes we should call k8s.